engineyard/ey-cookbooks-stable-v6
# [FB-2830] Adds letsencrypt

**Description of your patch**

Adds support for automated LetsEncrypt recipe using nothing but environmental variables with the exception of wildcard. 

**Recommended Release Notes**

LetsEncrypt support added via chef

**Estimated risk**

High risk modifies multiple components 

**Components involved**

nginx, ey-base, ey-init

**Dependencies**

None

**Description of testing done**

* Booted up fresh Environment with LetsEncrypt disabled / enabled
* Set certificate that wasn't generated by LetsEncrypt
* Set up LetsEncrypt
* Added instance to set up that was already running made sure keys were pushed correctly.

**QA Instructions**

* Check README.md in letsencrypt recipe
* Create environment with LetsEncrypt enabled
* Perform a take over
* Change between letsencrypt and normal certificate system
